src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
+    {
+        private bool _isHostShuttingDown;
+        private MessageReceiver _messageReceiver;
+        private readonly MessageHandlerOptions _messageHandlerOptions;
+        private readonly AzureServiceBusMessagePumpSettings _messagePumpSettings;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="serviceProvider">Collection of services that are configured</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        protected AzureServiceBusMessagePump(IConfiguration configuration, IServiceProvider serviceProvider,
+            ILogger logger)
+            : base(configuration, serviceProvider, logger)
+        {
+            _messagePumpSettings = serviceProvider.GetRequiredService<AzureServiceBusMessagePumpSettings>();
+            _messageHandlerOptions = DetermineMessageHandlerOptions(_messagePumpSettings);
+        }
+
+        /// <summary>
+        ///     Path of the entity to process
+        /// </summary>
+        public string EntityPath { get; private set; }
+
+        /// <summary>
+        ///     Service Bus namespace that contains the entity
+        /// </summary>
+        public string Namespace { get; private set; }
+
+        /// <inheritdoc />
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            _messageReceiver = await CreateMessageReceiverAsync(_messagePumpSettings);
```

Comment:
Should we add a sync lock arround field assigning?

Reply:
It's a singleton so I don't think so, or what's your concern?

Reply:
Ah, because this method could be called multiple times. To guard against corruption of the field. Or is that too far-fatched?

Reply:
I think it's goood, certainly for this point in time